core/src/main/scala/pureconfig/DerivedReaders.scala

```diff
@@ -40,17 +40,7 @@ trait DerivedReaders extends DerivedReaders1 {
 /**
  * Trait containing `ConfigReader` instances for collection, product and coproduct types.
  */
-trait DerivedReaders1 {
-
-  private[pureconfig] trait WrappedConfigReader[Wrapped, SubRepr] extends ConfigReader[SubRepr]
-
-  private[pureconfig] trait WrappedDefaultValue[Wrapped, SubRepr <: HList, DefaultRepr <: HList] {
-    def fromWithDefault(config: ConfigValue, default: DefaultRepr): Either[ConfigReaderFailures, SubRepr] = config match {
-      case co: ConfigObject => fromConfigObject(co, default)
-      case other => fail(WrongType(other.valueType, Set(ConfigValueType.OBJECT), ConfigValueLocation(other), ""))
-    }
-    def fromConfigObject(co: ConfigObject, default: DefaultRepr): Either[ConfigReaderFailures, SubRepr]
-  }
+trait DerivedReaders1 extends DerivedReaders2 {
```

Comment:
It's a pattern commonly used in libraries like scalaz and cats to deal with several layers of implicits. That means the implicits in `DerivedReaders` get higher priority, followed by `DerivedReaders1` (since `DerivedReaders` includes `DerivedReaders1`), followed by `DerivedReaders2`. They could be `DerivedReaders`, `LowerPriorityDerivedReaders` and `LowererPriorityDerivedReaders` :)

I had to do this because both `deriveOption` and `deriveCoproductInstance` can be used to derive a reader for `Option`, and the macro would see both of them as equal choices.